pyro/infer/mcmc/hmc.py

```diff
@@ -75,7 +75,7 @@ def _get_trace(self, z):
         return trace_poutine.trace
 
     def _kinetic_energy(self, r):
-        return 0.5 * torch.sum(torch.stack([r[name]**2 for name in r]))
+        return 0.5 * torch.sum(torch.cat([r[name]**2 for name in r]))
```

Comment:
Can we just change this to `0.5 * torch.sum(torch.cat([r[name].view(-1)**2 for name in r]))`? That should work with arbitrary sized tensors.

Reply:
Actually both of these look memory inefficient. Is there a good reason not to use
```py
return 0.5 * sum(x**2 for x in r.values())
```

Reply:
hey @fritzo, I removed your comment by mistake. Wanted to thumbs up. That was a good idea!

Reply:
@fritzo's comment:
Actually both of these look memory inefficient. Is there a good reason not to use
```
return 0.5 * sum(x**2 for x in r.values())
```